### PR TITLE
Lu prototype

### PR DESCRIPTION
**Pull request by felix642**
Added the prototype for LU_decomposition and finite differences. 

We need to wait for the LU Decomposition algorithm to be merged first.

### CODE REVIEW

File: prototype/large_scale_recommender/large_scale_recommender.ino [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+{
+  readElement(&feedbackSize, sizeof(feedbackSize));
+  const uint16_t totalSize = feedbackSize * feedbackSize;
+
+  feedbackMatrix = malloc(totalSize * sizeof(double));
+
+  for(uint8_t i = 0; i < totalSize; ++i)
+  {
+    readElement(&feedbackMatrix[i], sizeof(double));
+  }
```
Comment (bisson2000) [inline prototype/large_scale_recommender/large_scale_recommender.ino:81]: Could `readArray` work here?

File: prototype/large_scale_recommender/large_scale_recommender.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    print(f"We need to send a matrix of {size_side}x{size_side} = {nb_data} values")
+
+    for val in dataset:
+        port.write(struct.pack('<f', val))
+    return
+
+
+def main():
+    port = serial.Serial('/dev/ttyACM0', 9600)
+    time.sleep(2)
```
Comment (bisson2000) [inline prototype/large_scale_recommender/large_scale_recommender.py:66]: Could the custom serial class be used instead?

File: prototype/large_scale_recommender/large_scale_recommender.ino [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        sum += weight * currentSum * currentSum;
+      }
+    }
+  }
+  return fabs(sum);
+}
+
+static void dfunc(gradient_real* parameters, gradient_real* gradient)
+{
+  gradientApproximation(func, parameters, gradient, feedbackSize * LATENT_SIZE * 2, 0.0001, Central);
```
Comment (florenceCloutier) [inline prototype/large_scale_recommender/large_scale_recommender.ino:63]: Maybe explain what the 2 and the 0.0001 are in a comment or define as constants?

File: prototype/large_scale_recommender/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Large Scale Recommender
+
+## Presentation
+
+This prototype tries to solve the matrix factorization problem used in some large scale recommendation system. To do so,
```
Comment (nicholaslegrand102) [inline prototype/large_scale_recommender/README.md:5]: systems*?

**Review by zedfly — APPROVED** [review]
LGTM

**Review by nicholaslegrand102 — APPROVED** [review]
Looks good to me

**Review by bisson2000 — APPROVED** [review]
GFM

**Review by florenceCloutier — APPROVED** [review]
LGTM!

**Pull request merged**